File: typescript/base-hmi-types/src/screens/controls/HmiAlarmConnectionBindings.ts
import { HmiProperty } from "../base.js";

export class HmiAlarmConnectionBindings {
  acknowledgeNotification?: HmiProperty<number>;
  silenceNotification?: HmiProperty<number>;
  statusResetNotification?: HmiProperty<number>;
  messageNotification?: HmiProperty<number>;
  messageHandshake?: HmiProperty<number>;
  closeDisplayNotification?: HmiProperty<number>;
  remoteAcknowledge?: HmiProperty<number>;
  remoteAcknowledgeAll?: HmiProperty<number>;
  remoteAcknowledgeHandshake?: HmiProperty<number>;
  remoteCloseDisplay?: HmiProperty<number>;
  remoteClearHistory?: HmiProperty<number>;
  remoteSilence?: HmiProperty<number>;
  remoteStatusReset?: HmiProperty<number>;
}

File: src/BaseHmiTypes/Screens/Controls/HmiAlarmConnectionBindings.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

/// <summary>Inert tag and expression bindings used by classic alarm controls.</summary>
public class HmiAlarmConnectionBindings
{
    public HmiProperty<double>? AcknowledgeNotification { get; set; }

    public HmiProperty<double>? SilenceNotification { get; set; }

    public HmiProperty<double>? StatusResetNotification { get; set; }

    public HmiProperty<double>? MessageNotification { get; set; }

    public HmiProperty<double>? MessageHandshake { get; set; }

    public HmiProperty<double>? CloseDisplayNotification { get; set; }

    public HmiProperty<double>? RemoteAcknowledge { get; set; }

    public HmiProperty<double>? RemoteAcknowledgeAll { get; set; }

    public HmiProperty<double>? RemoteAcknowledgeHandshake { get; set; }

    public HmiProperty<double>? RemoteCloseDisplay { get; set; }

    public HmiProperty<double>? RemoteClearHistory { get; set; }

    public HmiProperty<double>? RemoteSilence { get; set; }

    public HmiProperty<double>? RemoteStatusReset { get; set; }
}
